infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -56,27 +80,119 @@ def run_shell_command(command, env_vars):
+
+
+def upload_backup_to_s3(
+    s3_client: BaseClient,
+    backup_fpath: Path,
+    backup_bucket_name: str,
+    backup_object_name: str,
+):
+    """Uploads the backup_fpath file to AWS S3.
+
+    :param s3_client: an S3 client to use to upload the file, defaults to None
+    :param backup_fpath: the filepath to the file that will be uploaded
+        , defaults to None
+    :param backup_bucket_name: the S3 bucket to upload the file to
+        , defaults to None
+    :param backup_object_name: the filename for the uploaded object in S3
+        , defaults to None
+    """
+    with open(str(backup_fpath), "rb") as f:
+        s3_client.upload_fileobj(Fileobj=f, Bucket=backup_bucket_name, Key=backup_object_name)
+
+
+def delete_local_backup_file(backup_fpath: Path):
+    """Deletes the local backup file once it has been uplaoded to S3.
+
+    :param backup_fpath: the filepath to the file that will be deleted
+        , defaults to None
+    """
+    os.remove(str(backup_fpath))
+
 
+def backup_database(backup_object_name: str):
+    """Creates a local backup of the database, uploades it to S3, and delets the local backup.
 
-def backup_database(db_backup_gzip_fpath):
+    :param backup_object_name: the name to be used for the backup file
+        , defaults to None
+    """
     # make sure the backup directory exists
+    db_backup_gzip_fpath = make_backup_fpath(object_name=backup_object_name)
     db_backup_gzip_fpath.parent.mkdir(parents=True, exist_ok=True)
 
     # backup the database
-    print("Backing up database to", db_backup_gzip_fpath)
+    print("Creating the backup")
 
     # run the backup command
-    backup_cmd = (
-        "pg_dumpall --dbname {conn_string} --clean --if-exists | gzip --stdout > {backup_fpath}".format(
-            conn_string=CONNECTION_STRING, backup_fpath=db_backup_gzip_fpath
-        )
+    backup_cmd = "pg_dumpall --dbname {conn_string} | gzip --stdout > {backup_fpath}".format(
+        conn_string=CONNECTION_STRING, backup_fpath=db_backup_gzip_fpath
     )
-    run_shell_command(backup_cmd, env_vars={"PGPASSWORD": os.environ["POSTGRES_PASSWORD"]})
+    run_shell_command(command=backup_cmd, env_vars={"PGPASSWORD": os.environ["POSTGRES_PASSWORD"]})
+
+    # upload the backup to S3
+    print("Backing up the database as", backup_object_name, "to S3")
+    s3_client = create_s3_client()  # s3_client: BaseClient = create_s3_client()
+    upload_backup_to_s3(
+        s3_client=s3_client,
+        backup_fpath=db_backup_gzip_fpath,
+        backup_bucket_name=BACKUP_BUCKET,
+        backup_object_name=backup_object_name,
+    )
+
+    # delete local backup
+    delete_local_backup_file(backup_fpath=db_backup_gzip_fpath)
 
 
-def backup_database_on_interval(seconds):
+def backup_database_on_interval(seconds: Union[int, float]):
+    """Backs up the database to S3 repeatedly on an interval.
+
+    :param seconds: the number of seconds to wait inbetween backups
+        , defaults to None
```

Comment:
Same comment as earlier. If it defaulted to `None`, the signature would be `seconds: Optional[Union[int, float]] = None`